### ISSUE TITLE
reviewing a PR

### ISSUE THREAD

**Issue by simonpcouch** [RhoBott/data_at_reed#10]
## wheeee

okay okay okay! the last step of getting the hang of git-things before we can jump into really grooving on new content is reviewing pull requests.

reviewing PRs is as much a social thing as it is a technical thing, and doing it "well" largely depends on context. you're taking a look over someones code, providing feedback, and deciding whether it's a good idea to merge their changes into "prod" (as in, the production version)—for us, prod is `main`.

for us, it's probably _always_ a good idea to merge other people's changes. (we're good at what we do, and only have a few hours a week to focus on this.) 

to look at what someone changed in a PR, we'll check out #9 from Leila as an example. if you tab over to "Files Changed", you'll see the "diff" of her work. (as in, what's different?) diffs are your friend! they show you exactly what changed between the old and new version.

one helpful thing about github is that it will tell you 1) how many files changed and 2) by how much did they changed. for #9, 1 file changed, and `+99 −0` (up in the top right corner in green and red!) tells us that she _only_ added stuff.

to provide feedback, you can either just comment on the PR generally, or, when looking at the diff, clicking specific lines—a little comment box will pull up. josh did this on #9

---------------

## okay okay get to the point

we're going to get some practice reviewing each other's PRs!

sometime this week:

look for the most recent PR, linked to this issue by dropping #10 in the title (or anywhere in the PR description)! and then..

#### review the PR 

* read the author's description
* look at the code
* maybe comment a bit
* maybe change some things by committing directly to the branch that the author PRed from

#### and then merge it

* to merge someone's PR, click "merge pull request".
* adjust the commit message to link to this issue (#10)
* delete the branch that was merged (a merged branch is usually called "stale", i.e. it will only confuse people to keep around)

#### branch off and do some cool stuff!

* go back to the [main @dr page](https://github.com/simonpcouch/data_at_reed)
* **make a fresh new branch off of `main`** (rather than moving back to the branch you were working on previosuly)
* pull in rstudio (big blue down arrow)
* switch to that branch (to the right of the blue arrow, a text drop-down)--use the `REMOTE: ORIGIN` option rather than `LOCAL BRANCHES`!
* pull again (big blue down arrow)
*  find something you want to work on and make some changes + commit them. 

#### make a new PR

* PR your changes back to `main` (instructions in the [old README](https://github.com/simonpcouch/data_at_reed/tree/main/readmes/v1)) 
* make sure to describe what you did and link to any relevant issues (+ this one, possibly).
* wait for whoever is on shift next to merge your PR!

person 1 will review my PR, person 2 will review person 1's, etc :-)

[EDIT: once you've made the PR, there will be a little gray `#number` next to your PR. this is a label that git/github assigns to your PR/issue so that it can be referenced/linked to later on! now that i've made this issue... "reviewing a PR" is the title, #10 is the "label" that github assigned to it, the thing you're reading is the "description". both PRs and issues get each of these three, see #12 for an example of how it can look on a PR.]
